clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -177,6 +177,7 @@ createTargetCodeGenInfo(CodeGenModule &CGM) {
     else if (CodeGenOpts.FloatABI == "hard" ||
              (CodeGenOpts.FloatABI != "soft" &&
               (Triple.getEnvironment() == llvm::Triple::GNUEABIHF ||
```

Comment:
I'm wondering if we should add a `isHardFloatABI()` or something like that to the `Triple` class.

Reply:
This helper sounds good.

will we have `muslt32`?

Reply:
Are you asking if we're going to have musl suffixes on Gentoo? No, musl already switched to 64-bit `time_t`.